src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,154 @@
+     * @param file The file name
+     */
+    public Storage(String fileDir, String file) {
+        this.directory = fileDir;
+        this.file = file;
+        this.fileTasks = new ArrayList<>();
+    }
+
+    /**
+     * reads file. If file does not exist, create file
+     * @return List containing all tasks in file originally
+     */
+    public ArrayList<Task> loadData() throws DukeException {
+        // Make directory and/or file if they don't exist
+        File dataFolder = new File(directory);
+        boolean folderCreated = dataFolder.mkdirs();
+        File dataFile = new File(directory + "/" + file);
+        try {
+            boolean fileCreated = dataFile.createNewFile();
+        } catch (IOException e) {
+            System.out.println("Failed to create a new file");
+            return null;
+        }
+
+        // list to contain all tasks ass understood by bot.
+        ArrayList<Task> fileList = new ArrayList<>();
+        try {
+            Scanner fileReader = new Scanner(dataFile);
+            fileTasks.clear();
+            while (fileReader.hasNextLine()) {
+                String rawData = fileReader.nextLine();
+                fileTasks.add(rawData);
+                String[] data = rawData.split(" \\| ");
+                String taskType = data[0];
+                boolean isDone = data[1].equals("1");
+                Task task = null;
+                switch (taskType) {
+                    case "T":
+                        // Add a todo task.
+                        task = new Todo(data[2]);
+
+                        break;
+                    case "D":
+                        // Add a deadline task.
+                        task = new Deadline(data[2], data[3]);
+
+                        break;
+                    case "E":
+                        // Add an event task.
+                        task = new Event(data[2], data[3]);
+                        break;
+                }
+                if (task != null) {
+                    if (isDone) {
+                        task.doneTask();
+                    }
+                    fileList.add(task);
+                }
+            }
+        } catch (FileNotFoundException e) {
+            throw new DukeException("No saved data found");
+        }
+        return fileList;
+    }
+
+    /**
+     * Adds a task to the saved file
+     * @param task The task as a String
+     * @throws DukeException When saving the file fails
+     */
+    public void addToFile(String task) throws DukeException, IOException {
+        fileTasks.add(task);
+        saveToFile();
+    }
+
+    /**
+     * Removes a task from the saved file
+     * @param id The line to be removed
+     * @throws DukeException When saving the file fails
+     */
+    public void deleteFromFile(int id) throws DukeException, IOException {
+        fileTasks.remove(id - 1);
+        saveToFile();
+    }
+
+    /**
+     * updates a line in the file at the given index
```

Comment:
Perhaps you can capitalize the U, add a full stop, and have an empty line between your header and param.